Probe/src/probe/logger.go

```diff
@@ -16,21 +16,79 @@
 
 package probe
 
-import "log"
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+)
 
+// Wrapper for logging
 type Logger interface {
 	LogProbe(sp *sentProbe, st string, lat int)
+	LogError(desc string)
+	LogFatal(desc string)
+	LogFatalf(desc string, args ...interface{})
 }
 
+// Logger that sends logs to cloud logger via gcloud
 type CloudLogger struct {
-	// TODO(langenbahn): Add fields when logging is implemented
 }
 
-func NewCloudLogger() *CloudLogger {
-	return new(CloudLogger)
+//TODO(langenbahn): add log/error destinations in config proto
+type probeLog struct {
+	sendTime  string
+	probeType string
+	latency   int
+	state     string
+	region    string
+	token     string
 }
 
+type errorLog struct {
+	desc   string
+	region string
+	token  string
+}
+
+// Log probe information to specified log
 func (c *CloudLogger) LogProbe(sp *sentProbe, st string, lat int) {
-	//TODO(langenbahn): Implement logging
-	log.Printf("time: %s status: %s latency: %d", sp.sendTime.Format(timeLogFormat), st, lat)
+	cl := &probeLog{sp.sendTime.Format(timeLogFormat), sp.probe.getTypeString(), lat, st,
+		sp.probe.config.GetRegion(), deviceToken}
+	l, err := json.Marshal(cl)
+	if err != nil {
+		c.LogError(fmt.Sprintf("Unable to log probe: unable to marshal JSON: %v", err))
+	}
+
+	err = maker.Command("gcloud", "logging", "write",
+		"--payload-type=json", "PROBELOG", string(l)).Run()
+	if err != nil {
+		c.LogError(fmt.Sprintf("Unable to log probe: unable to send to server: %v", err))
+	}
+}
+
+func (c *CloudLogger) LogFatal(desc string) {
+	c.LogError(fmt.Sprintf("fatal: %s", desc))
+	os.Exit(1)
+}
+
+func (c *CloudLogger) LogFatalf(desc string, args ...interface{}) {
+	c.LogError(fmt.Sprintf("fatal: "+desc, args...))
+	os.Exit(1)
+}
+
+// Log errors to specified log
+func (c *CloudLogger) LogError(desc string) {
+	//TODO(langenbahn) add any other useful information for errors
+	el := &errorLog{desc, probeConfigs.Probe[0].GetRegion(), deviceToken}
+	l, err := json.Marshal(el)
+	if err != nil {
+		log.Printf("Unable to log error: unable to marshal JSON: %v", err)
```

Comment:
You should exit the method here, otherwise you're using l below even though it's broken

Reply:
Addressed in c9ccc1a